sda/cmd/api/api_test.go

```diff
@@ -0,0 +1,511 @@
+	assert.Equal(t, http.StatusServiceUnavailable, res.StatusCode)
+	defer res.Body.Close()
+
+	// reconnect should be fast so now this should pass
+	res, err = http.Get(ts.URL + "/ready")
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusOK, res.StatusCode)
+	defer res.Body.Close()
+
+	// close the channel to force a reconneciton
+	Conf.API.MQ.Channel.Close()
+	res, err = http.Get(ts.URL + "/ready")
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusServiceUnavailable, res.StatusCode)
+	defer res.Body.Close()
+
+	// reconnect should be fast so now this should pass
+	res, err = http.Get(ts.URL + "/ready")
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusOK, res.StatusCode)
+	defer res.Body.Close()
+
+	// close DB connection to force a reconnection
+	Conf.API.DB.Close()
+	res, err = http.Get(ts.URL + "/ready")
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusServiceUnavailable, res.StatusCode)
+	defer res.Body.Close()
+
+	// reconnect should be fast so now this should pass
+	res, err = http.Get(ts.URL + "/ready")
+	assert.NoError(t, err)
+	assert.Equal(t, http.StatusOK, res.StatusCode)
+	defer res.Body.Close()
+}
+
+type TestSuite struct {
+	suite.Suite
+	Path        string
+	PublicPath  string
+	PrivatePath string
+	KeyName     string
+	Token       string
+	User        string
+}
+
+func TestApiTestSuite(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+// Initialise configuration and create jwt keys
+func (suite *TestSuite) SetupTest() {
+
+	suite.Path = "/tmp/keys/"
+	suite.KeyName = "example.demo"
+
+	log.Print("Creating JWT keys for testing")
+	privpath, pubpath, err := helper.MakeFolder(suite.Path)
+	assert.NoError(suite.T(), err)
+	suite.PrivatePath = privpath
+	suite.PublicPath = pubpath
+	err = helper.CreateRSAkeys(privpath, pubpath)
+	assert.NoError(suite.T(), err)
+
+	// Create a valid token for queries to the API
+	prKeyParsed, err := helper.ParsePrivateRSAKey(suite.PrivatePath, "/rsa")
+	assert.NoError(suite.T(), err)
+	token, err := helper.CreateRSAToken(prKeyParsed, "RS256", helper.DefaultTokenClaims)
+	assert.NoError(suite.T(), err)
+	suite.Token = token
+	user, ok := helper.DefaultTokenClaims["sub"].(string)
+	assert.True(suite.T(), ok)
+	suite.User = user
+
+	c := &config.Config{}
+	ServerConf := config.ServerConfig{}
+	ServerConf.Jwtpubkeypath = suite.PublicPath
+	c.Server = ServerConf
+
+	Conf = c
+
+	log.Print("Setup DB for my test")
+	Conf.Database = database.DBConf{
+		Host:     "localhost",
+		Port:     dbPort,
+		User:     "lega_in",
+		Password: "lega_in",
+		Database: "lega",
+		SslMode:  "disable",
+	}
+	Conf.API.DB, err = database.NewSDAdb(Conf.Database)
+	assert.NoError(suite.T(), err)
+
+}
+
+func TestDatabasePingCheck(t *testing.T) {
+	database := database.SDAdb{}
+	assert.Error(t, checkDB(&database, 1*time.Second), "nil DB should fail")
+
+	database.DB, _, err = sqlmock.New()
+	assert.NoError(t, err)
+	assert.NoError(t, checkDB(&database, 1*time.Second), "ping should succeed")
```

Comment:
```suggestion
	db, err := database.NewSDAdb(Conf.Database)
	assert.NoError(suite.T(), err)
	assert.NoError(suite.T(), checkDB(db, 1*time.Second), "ping should succeed")
```

You have a real database, no need for any mock instances.